src/cursor/cur_std.c

```diff
@@ -1290,6 +1290,38 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
     API_END_RET_STAT(session, ret, cursor_bound);
 }
 
+/*
+ * __wt_cursor_bounds_save --
+ *     Save cursor bounds to restore the state.
+ */
+void
+__wt_cursor_bounds_save(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *state)
+{
+    WT_ITEM_SET(state->lower_bound, cursor->lower_bound);
+    WT_ITEM_SET(state->upper_bound, cursor->upper_bound);
+    state->flags = cursor->flags;
```

Comment:
I would repurpose and rename the state->flags to state->bound_flags and fetch only the bound flags here.

Reply:
[3ae3da2](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/3ae3da25e2566a302ea82fce4a8d45b87daa8bcf)